amun/core.py

```diff
@@ -66,11 +66,23 @@ def create_inspect_buildconfig(openshift: OpenShift, inspection_id: str, dockerf
     _LOGGER.debug("OpenShift response for getting Amun inspect BuildConfig template: %r", response)
 
     template = response['items'][0]
-    openshift.set_template_parameters(
-        template,
-        AMUN_INSPECTION_ID=inspection_id,
-        AMUN_GENERATED_DOCKERFILE=dockerfile
-    )
+    if 'build' in specification:
```

Comment:
You provided defaults in the template. If you do not provide `AMUN_BUILD_CPU` and `AMUN_BUILD_MEMORY` to `set_template_parameters` the default values should be used. So this if statement is not necessary; here and in the job function as well.

Reply:
If we do not have the IF condition then this line 
        build_specification = specification['build']['response']
will throw an error or it will pass null parameters in set template parameters